visualization/app/codeCharta/ui/searchBar/searchBar.component.spec.ts

```diff
@@ -91,6 +91,46 @@ describe("SearchBarController", () => {
 		})
 	})
 
+	describe("onClickBlacklistPattern", () => {
+		it("separate entries for many in one searchPattern", () => {
+			const blacklistItem = { path: "*html*", type: BlacklistType.exclude }
+			const blacklistItem1 = { path: "*ts*", type: BlacklistType.exclude }
+			searchBarController["_viewModel"].searchPattern = "html,ts"
+			storeService.dispatch(setSearchPattern(blacklistItem.path))
+
+			searchBarController.onClickBlacklistPattern(BlacklistType.exclude)
+
+			expect(storeService.getState().fileSettings.blacklist).toContainEqual(blacklistItem)
+			expect(storeService.getState().fileSettings.blacklist).toContainEqual(blacklistItem1)
+			expect(searchBarController["_viewModel"].searchPattern).toBe("")
+			expect(storeService.getState().dynamicSettings.searchPattern).toBe("")
+
+			searchBarController.onSearchPatternChanged("ts")
+			expect(searchBarController["_viewModel"].isPatternHidden).toBeFalsy()
+			expect(searchBarController["_viewModel"].isPatternExcluded).toBeTruthy()
+		})
+	})
+
+	describe("onClickBlacklistPattern", () => {
+		it("separate entries for many in one searchPattern", () => {
+			const blacklistItem = { path: "!*html*", type: BlacklistType.exclude }
+			const blacklistItem1 = { path: "!*ts*", type: BlacklistType.exclude }
+			searchBarController["_viewModel"].searchPattern = "!html,ts"
+			storeService.dispatch(setSearchPattern(blacklistItem.path))
+
+			searchBarController.onClickBlacklistPattern(BlacklistType.exclude)
+
+			expect(storeService.getState().fileSettings.blacklist).toContainEqual(blacklistItem)
+			expect(storeService.getState().fileSettings.blacklist).toContainEqual(blacklistItem1)
+			expect(searchBarController["_viewModel"].searchPattern).toBe("")
+			expect(storeService.getState().dynamicSettings.searchPattern).toBe("")
+
+			searchBarController.onSearchPatternChanged("!ts")
```

Comment:
I think we should add some more tests for other search patterns. But the ideal way would be to reuse `CodeMapHelper.getNodesByGitignorePath`

Reply:
I think that the more tests the better. but the said method  is used for searching and displaying the buildings and not in the exclusion of the building. The exclusion of buldings has been tested in the decorate map function in nodeDecorator and I think I have added a test concerning it there in the task 901 to cover more the exclusion of nodes.